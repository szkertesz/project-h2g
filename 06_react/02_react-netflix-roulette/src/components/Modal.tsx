import { useEffect, useRef } from 'react';
import { CSSTransition } from 'react-transition-group';
import Portal from './Portal';
import classes from './Modal.module.scss';

interface Props {
    children: JSX.Element;
    isOpen: boolean;
    handleClose: () => void;
}

const Modal: React.FC<Props> = ({ children, isOpen, handleClose }) => {
    const nodeRef = useRef(null);

    useEffect(() => {
        const closeOnEscapeKey = (e: KeyboardEvent) =>
            e.key === 'Escape' ? handleClose() : null;
        document.body.addEventListener('keydown', closeOnEscapeKey);
    });
    // if (!isOpen) return null;

    return (
        <Portal containerId='portal-container'>
            <CSSTransition
                in={isOpen}
                timeout={{ entry: 0, exit: 300 }}
                unmountOnExit
                classNames={{
                    enterDone: classes['modal-enter-done'],
                    exit: classes['modal-exit']
                }}
                nodeRef={nodeRef}
            >
                <div onClick={handleClose} ref={nodeRef} className={classes.modal}>
                    <div
                        onClick={(e) => e.stopPropagation()}
                        className={classes['modal__dialog']}
                    >
                        <button
                            onClick={handleClose}
                            className={classes['modal__close']}
                        >
                            Close
                        </button>
                        <div className={classes['modal__content']}>
                            {children}
                        </div>
                    </div>
                </div>
            </CSSTransition>
        </Portal>
    );
};

export default Modal;
