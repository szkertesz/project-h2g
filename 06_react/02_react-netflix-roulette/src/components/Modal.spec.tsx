import React from 'react';
import Modal from './Modal';
import { render, screen } from '@testing-library/react';

describe('Modal', () => {
    describe('should render', () => {
        it('component', () => {
            const children = `<p>Modal child</p>`
            const isOpen = true
            const size = 'small'
            const handleClose = jest.fn()
            const title = 'modal title'

            const { container } = render(
                <Modal
                    children={children}
                    isOpen={isOpen}
                    size={size}
                    handleClose={handleClose}
                    title={title}
                />
            );
            expect(container).toMatchSnapshot();
        });
    });
});
