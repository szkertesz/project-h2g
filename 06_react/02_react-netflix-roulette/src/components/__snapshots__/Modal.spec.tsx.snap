// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render component 1`] = `<div />`;
